src/frontend/src/handler/create_source.rs

```diff
@@ -268,39 +279,70 @@ pub(crate) async fn resolve_source_schema(
         },
 
         SourceSchema::DebeziumAvro(avro_schema) => {
-            if row_id_index.is_some() {
-                return Err(RwError::from(ProtocolError(
-                    "Primary key must be specified when creating table with row format
-            debezium_avro."
-                        .to_string(),
-                )));
-            }
-
-            if columns.len() != pk_column_ids.len() {
-                return Err(RwError::from(ProtocolError(
-                    "User can only specify primary key columns when creating table with row
-            format debezium_avro."
-                        .to_string(),
-                )));
-            }
-
-            let mut full_columns =
-                extract_debezium_avro_table_schema(avro_schema, with_properties.clone()).await?;
-
-            for pk_column in columns.iter() {
-                let index = full_columns
-                    .iter()
-                    .position(|c| c.column_desc.name == pk_column.column_desc.name)
-                    .ok_or_else(|| {
-                        RwError::from(ProtocolError(format!(
-                            "pk column {} not exists",
-                            pk_column.column_desc.name
-                        )))
-                    })?;
-                let _ = full_columns.remove(index);
+            // no row_id, so user specify pk(s)
+            if row_id_index.is_none() {
+                // user specify no only pk columns
+                if columns.len() != pk_column_ids.len() {
+                    return Err(RwError::from(ProtocolError(
+                        "User can only specify primary key columns when creating table with row
+                        format debezium_avro."
+                            .to_owned(),
```

Comment:
so 
```sql
create source s(xxx int, yyy int, ...) with ( ... ) row format debezium_avro schema registry http://xxx
```
is not allowed?

what if user specified columns are not consistent with columns in schema registry

Reply:
> so
> 
> ```sql
> create source s(xxx int, yyy int, ...) with ( ... ) row format debezium_avro schema registry http://xxx
> ```
> 
> is not allowed?
> 

Yes, user can only specify pk column(s) or nothing.

> what if user specified columns are not consistent with columns in schema registry

We use the pk column names specified by user to find real pk_column_ids in schema extracted from schema registry, if they do not match, the program will error.